op-node/rollup/derive/plasma_data_source.go

```diff
@@ -29,21 +31,61 @@ func NewPlasmaDataSource(log log.Logger, src DataIter, fetcher PlasmaInputFetche
 }
 
 func (s *PlasmaDataSource) Next(ctx context.Context) (eth.Data, error) {
+	// Process origin syncs the challenge contract events and updates the local challenge states
+	// before we can proceed to fetch the input data. This function can be called multiple times
+	// for the same origin and noop if the origin was already processed. It is also called if
+	// there is not commitment in the current origin.
+	if err := s.fetcher.AdvanceL1Origin(ctx, s.id); err != nil {
+		if errors.Is(err, plasma.ErrReorgRequired) {
+			return nil, NewResetError(fmt.Errorf("new expired challenge"))
+		}
+		return nil, NewTemporaryError(fmt.Errorf("failed to advance plasma L1 origin: %w", err))
+	}
+
 	if s.comm == nil {
 		var err error
 		// the l1 source returns the input commitment for the batch.
-		s.comm, err = s.src.Next(ctx)
+		data, err := s.src.Next(ctx)
 		if err != nil {
 			return nil, err
 		}
+		// validate batcher inbox data is a commitment.
+		comm, err := plasma.DecodeKeccak256(data)
+		if err != nil {
+			s.log.Warn("invalid commitment", "data", fmt.Sprintf("%x", data), "err", err)
```

Comment:
```suggestion
			s.log.Warn("invalid commitment", "commitment", data, "err", err)
```
hexutils.Bytes already has a string method. It was previously getting double encoded (i.e. bytes -> hex string -> re-interpreted as bytes -> second round of hex string).

Reply:
Thanks, fixed!